addons/website_sale_stock/static/src/js/website_sale.js

```diff
@@ -0,0 +1,53 @@
+/** @odoo-module alias=website_sale_stock.website_sale**/
+
+import {WebsiteSale} from 'website_sale.website_sale';
+import {is_email} from 'web.utils';
+
+WebsiteSale.include({
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        'click #stock_notification_product_message': '_onClickStockNotificationProductMessage',
+        'click #stock_notification_product_form_button': '_onSubmitStockNotificationProductForm',
+    }),
+
+    _onClickStockNotificationProductMessage: function (ev) {
+        ev.currentTarget.classList.add('d-none');
+        const partnerEmail = document.querySelector('#wsale_user_email').value;
+        const formEl = ev.currentTarget.parentElement.querySelector('#stock_notification_product_form');
+        const emailInputEl = formEl.querySelector('input[name="email"]');
+
+        emailInputEl.value = partnerEmail;
+        formEl.classList.remove('d-none');
+    },
+
+    _onSubmitStockNotificationProductForm: function (ev) {
+        const formEl = ev.currentTarget.closest('#stock_notification_product_form');
+        const productId = parseInt(formEl.querySelector('input[name="product_id"]').value);
+        const email = formEl.querySelector('input[name="email"]').value.trim();
+        const incorrectIconEl = formEl.querySelector('#stock_notification_product_input_incorrect');
+
+        if (!is_email(email)) {
+            incorrectIconEl.classList.remove('d-none');
+            return
+        }
+
+        this._rpc({
+            route: "/shop/add/stock_notification",
+            params: {
+                product_id: productId,
+                email,
+            },
+        }).then(function (data) {
+            if (data && 'error' in data) {
+                incorrectIconEl.classList.remove('d-none');
+                return
+            }
+            const div = formEl.closest('#stock_notification_product_div');
+            const message = div.querySelector('#stock_notification_product_success_message');
+
+            message.classList.remove('d-none');
+            formEl.classList.add('d-none');
+        });
```

Comment:
Since we now raise the error you can add a `.catch` clause to your rpc, also arrow function :)
```suggestion
        this._rpc({
            route: "/shop/add/stock_notification",
            params: {
                product_id: productId,
                email,
            },
        }).then((data) => {
            const div = formEl.closest('#stock_notification_product_div');
            const message = div.querySelector('#stock_notification_product_success_message');

            message.classList.remove('d-none');
            formEl.classList.add('d-none');
        }).catch((err) => {
            incorrectIconEl.classList.remove('d-none');
        });
```